src/pages/index.js

```diff
@@ -32,29 +45,87 @@ editFormValidator.enableValidation();
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
-function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo({
-    name: inputValues.title,
-    job: inputValues.description,
+// Get User Info
+
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo({
+      name: res.name,
+      job: res.about,
+    });
+    userInfo.setUserAvatar({ avatar: res.avatar });
+  })
+  .catch((err) => {
+    console.error(err);
   });
+
+// Update User Info
+
+function handleProfileEditSubmit(inputValues) {
+  profileEditPopup.setButtonText("Saving...");
+  api
+    .updateUserInfo(inputValues.title, inputValues.description)
+    .then(() => {
+      userInfo.setUserInfo({
+        name: inputValues.title,
+        job: inputValues.description,
+      });
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileEditPopup.setButtonText("Save");
+    });
   profileEditPopup.close();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well